common/sched_queue.go

```diff
@@ -0,0 +1,131 @@
+package common
+
+import (
+	"container/heap"
+	"math"
+	"time"
+
+	"github.com/benbjohnson/clock"
+)
+
+// A callable is a function that is called and return an (optional) next scheduled time
+type callable func() time.Time
+
+type schedCall struct {
+	t time.Time
+	c callable
+}
+
+func (sc *schedCall) FromNow(now time.Time) time.Duration {
```

Comment:
What do you think about flipping the receiver and arg types around, and changing the name? Then it would read
````
clock.Now().DurationUntil(scheduledCall)
````